src/python-common/ceph/deployment/hostspec.py

```diff
@@ -28,20 +29,22 @@ def __init__(self,
         #: human readable status
         self.status = status or ''  # type: str
 
+        self.maintenance = maintenance
+
     def to_json(self):
         return {
             'hostname': self.hostname,
             'addr': self.addr,
             'labels': self.labels,
-            'status': self.status,
+            'maintenance': self.maintenance,
         }
 
     @classmethod
     def from_json(cls, host_spec):
         _cls = cls(host_spec['hostname'],
                    host_spec['addr'] if 'addr' in host_spec else None,
                    host_spec['labels'] if 'labels' in host_spec else None,
-                   host_spec['status'] if 'status' in host_spec else None)
+                   host_spec['maintenance'] if 'maintenance' in host_spec else False)
```

Comment:
sure about this?

Reply:
we have to prevent `offline` from creeping into the config-key store.